webui/react/src/components/FilterForm/TableFilter.tsx

```diff
@@ -23,10 +25,27 @@ const TableFilter = ({
 }: Props): JSX.Element => {
   const columns: V1ProjectColumn[] = Loadable.getOrElse([], loadableColumns);
   const fieldCount = useObservable(formStore.fieldCount);
+  const formset = useObservable(formStore.formset);
 
-  const onIsOpenFilterChange = (newOpen: boolean) => {
-    setIsOpenFilter(newOpen);
-  };
+  const onIsOpenFilterChange = useCallback(
+    (newOpen: boolean) => {
+      if (newOpen) {
+        Loadable.match(formset, {
+          Loaded: (data) => {
+            // if there's no conditions, add default condition
+            if (data.filterGroup.children.length === 0) {
+              formStore.addChild(data.filterGroup.id, FormKind.Field);
+            }
+          },
+          NotLoaded: () => {
+            return;
+          },
+        });
+      }
+      setIsOpenFilter(newOpen);
+    },
+    [formStore, formset, setIsOpenFilter],
+  );
```

Comment:
Check if the filter is empty or not every time filter popup is open. If its empty, add a condition, otherwise, do nothing.